apps/electric_migrations/lib/electric_migrations/sqlite/parse.ex

```diff
@@ -0,0 +1,246 @@
+defmodule ElectricMigrations.Sqlite.Parse do
+  @moduledoc """
+  Creates an AST from SQL migrations
+  """
+  alias ElectricMigrations.Ast
+  alias ElectricMigrations.Sqlite.Introspect
+
+  @allowed_sql_types ["integer", "real", "text", "blob"]
+
+  @type migration :: %{original_body: String.t(), name: String.t()}
+
+  @doc """
+  Given a set of Maps and returns an ugly map of maps containing info about the DB structure.
+  Also validates the SQL and returns error messages if validation fails
+  """
+  @spec sql_ast_from_migrations([migration(), ...], String.t()) ::
+          {:ok, %{required(String.t()) => Ast.FullTableInfo.t()}, nil}
+          | {:error, [String.t(), ...]}
+  def sql_ast_from_migrations(migrations, default_namespace \\ "main") do
+    case ast_from_ordered_migrations(migrations, default_namespace) do
+      {ast, [], []} ->
+        {:ok, ast, nil}
+
+      {_ast, errors, _warnings} ->
```

Comment:
Because if there are errors, then we don't care about warnings. At least, that was the initial logic, but warnings never were added as far as I can tell, so  this is a dead field I didn't remove. Warnings as a concept is in a bunch of places despite not being used, so I left it in for now due to lack of time